File: BookNestFrontend/booknest_mobile/lib/models/event_category.dart
class EventCategory {
  final int id;
  final String name;
  final String description;

  EventCategory({
    required this.id,
    required this.name,
    required this.description,
  });

  factory EventCategory.fromJson(Map<String, dynamic> json) {
    return EventCategory(
      id: json['id'],
      name: json['name'] ?? '',
      description: json['description'] ?? '',
    );
  }
}

File: BookNestBackend/BookNest.Services/Database/Entities/EventCategory.cs
using System.ComponentModel.DataAnnotations;

namespace BookNest.Services.Database.Entities
{
    public class EventCategory
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        public ICollection<Event> Events { get; set; } = new List<Event>();
    }
}
